pkg/alertmanager/validation/v1beta1/validation.go

```diff
@@ -259,12 +259,20 @@ func validateVictorOpsConfigs(configs []monitoringv1beta1.VictorOpsConfig) error
 
 func validatePushoverConfigs(configs []monitoringv1beta1.PushoverConfig) error {
 	for _, config := range configs {
-		if config.UserKey == nil {
-			return errors.Errorf("mandatory field %q is empty", "userKey")
+		if config.UserKey == nil && config.UserKeyFile == nil {
+			return errors.Errorf("one of userKey or userKeyFile must be configured")
 		}
 
-		if config.Token == nil {
-			return errors.Errorf("mandatory field %q is empty", "token")
+		if config.UserKey != nil && config.UserKeyFile != nil {
+			return errors.Errorf("at most one of userKey & userKeyFile must be configured")
+		}
```

Comment:
We have something similar with the telegram receiver, we don't want to send validation errors if both UserKey and UserKeyFile are set, but when reconciling the alertmanager resource we ignore the field with lower prio

Reply:
Thanks for your review. Delete validation